### PR TITLE
ECC code mismatch detection

### PR DESCRIPTION
**Pull request by Shane32**
This pull request enhances the QRCoder library by allowing the payload to specify an "unspecified" error correction level (ECC) rather than being restricted to the predefined levels L, M, Q, or H. This new `Default` value in the `ECCLevel` enum acts as an unspecified placeholder, giving users the flexibility to select an appropriate ECC level while enabling the system to validate the selection.

When generating a QR code, if the ECC level is not specified, the system uses the payload's specified ECC level. If the payload's ECC level differs from the user-specified level, the system throws an exception to handle the discrepancy. This enhancement ensures backward compatibility, allowing existing codebases to function as expected while offering users the flexibility to dynamically adjust error correction levels based on their payloads' needs.

See:
- https://github.com/codebude/QRCoder/issues/512#issuecomment-2118994971

This PR also includes bug fix #527 ; please merge it first.

### CODE REVIEW

**Review by codebude — APPROVED** [review]

**Comment by codebude** [comment]
That's a nice solution. I wonder if we shouldn't just include it in the next 1.x.x release. I don't see any reason why we need to keep the change for version 2.

Just a small test case (something like generator_throws_argument_exception or similar) would be nice before merging it.

**Comment by Shane32** [comment]
This particular implementation is a breaking change since I’ve changed the type. But we could add an enum value like EccLevel.Default to do the same thing, which would be backwards compatible. I’ll revise.

File: QRCoderApiTests/net35+net40+net50+net50-windows+netstandard20/QRCoder.approved.txt [hunk 1]
```diff
@@ -895,6 +895,7 @@ namespace QRCoder
         public static QRCoder.QRCodeData GenerateQrCode(string plainText, QRCoder.QRCodeGenerator.ECCLevel eccLevel, bool forceUtf8 = false, bool utf8BOM = false, QRCoder.QRCodeGenerator.EciMode eciMode = 0, int requestedVersion = -1) { }
         public enum ECCLevel
         {
+            Default = -1,
             L = 0,
             M = 1,
             Q = 2,
```
Comment (Shane32) [inline QRCoderApiTests/net35+net40+net50+net50-windows+netstandard20/QRCoder.approved.txt:901]: Only API change is to add a new enum entry, ensuring compatibility with code built against prior versions.

File: QRCoder/PayloadGenerator.cs [hunk 2]
```diff
@@ -15,7 +15,7 @@ public static class PayloadGenerator
         public abstract class Payload
         {
             public virtual int Version { get { return -1; } }
-            public virtual QRCodeGenerator.ECCLevel EccLevel { get { return QRCodeGenerator.ECCLevel.M; } }
+            public virtual QRCodeGenerator.ECCLevel EccLevel { get { return QRCodeGenerator.ECCLevel.Default; } } // null is default
```
Comment (codebude) [inline QRCoder/PayloadGenerator.cs:18]:
```suggestion
            public virtual QRCodeGenerator.ECCLevel EccLevel { get { return QRCodeGenerator.ECCLevel.Default; } }
```
As default is -1 now, we don't need the comment any longer. (Mentioning that it is -1 isn't necessary as it's interna of the Enum.)

**Review by codebude — APPROVED** [review]

**Comment by csturm83** [comment]
Isn't it a breaking change to throw an exception when previously didn't? Might be worth mentioning in the release notes.

**Comment by codebude** [comment]
@csturm83 Thanks for pointing this out. Yes, you're right - this is somehow breaking behaviour but In my opinion nevertheless totally worth it, as cases when the exception will be raised, otherwise would lead to incorrect payloads/QR codes. (Thus I assume the change won't break nearly none implementation out in the wild, because to break it, the implementations must create in correct codes at the moment.)

I added a note on this change in the release notes: 
- https://github.com/codebude/QRCoder/wiki/Release-notes#breaking-changes

**Pull request merged**